### ISSUE TITLE
The inference takes 30 seconds per image on CPU and the precision is a little bad.

### ISSUE THREAD

**Issue by Honghe** [Delay-Xili/F-Clip#5]
Env:
- Ubuntu 20.04
- Python 3.8
- PyTorch 1.8.1

CPU:
```
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   39 bits physical, 48 bits virtual
CPU(s):                          4
On-line CPU(s) list:             0-3
Thread(s) per core:              1
Core(s) per socket:              4
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           94
Model name:                      Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz
Stepping:                        3
CPU MHz:                         3503.549
CPU max MHz:                     3600.0000
CPU min MHz:                     800.0000
```

Inference log:
```
$ CUDA_VISIBLE_DEVICES= python demo.py /data/ --model HG2 --output_dir logs/demo_result --ckpt /fclip_ckpt/ckpt/HG2/checkpoint.pth.tar --display False
Namespace(camid=0, ckpt='/fclip_ckpt/ckpt/HG2/checkpoint.pth.tar', display=True, img_glob='*.png', input='/data/', model='HG2', output_dir='logs/demo_result', skip=1, waitkey=1)
logs/demo_result
==> Loading video.
[ERROR:0] global /tmp/pip-req-build-rcdoave0/opencv/modules/videoio/src/cap.cpp (160) open VIDEOIO(CV_IMAGES): raised OpenCV exception:

OpenCV(4.5.2) /tmp/pip-req-build-rcdoave0/opencv/modules/videoio/src/cap_images.cpp:253: error: (-5:Bad argument) CAP_IMAGES: can't find starting number (in the name of file): /data/ in function 'icvExtractPattern'


==> Processing Image Directory Input.
==> Successfully loaded video.
Using device:  cpu
0-th head, head type raw, head output 2
1-th head, head type raw, head output 2
2-th head, head type raw, head output 1
3-th head, head type raw, head output 1
0-th head, head type raw, head output 2
1-th head, head type raw, head output 2
2-th head, head type raw, head output 1
3-th head, head type raw, head output 1
=> loading model from /fclip_ckpt/ckpt/HG2/checkpoint.pth.tar
Finished constructing model!
==> Running Demo.
Processing: 1 Total time spent:30.598598
Average frame rate:0.032681
==> Finshed Demo.
```
The origin image:
![origin image](https://user-images.githubusercontent.com/1092722/125078249-b51ab300-e0f4-11eb-9548-6ca386353902.png)

The output image:
![output image](https://user-images.githubusercontent.com/1092722/125078259-b9df6700-e0f4-11eb-87e4-c4b4293445b5.png)

**Comment by Honghe** [comment]
@Delay-Xili Is this inference elapsed-time normal?

**Comment by Delay-Xili** [comment]
Hi @Honghe , 
sorry for the late reply. 
Basically, the "Total time spent:30.598598" including three main time-consuming parts: 1. the line detection time. 2. line drawing time. 3. some other operators like to read images and save images. You can try to test the time of them separately. 
The model you used "HG2" can achieve 35 FPS on the server with one 2080Ti GPU. Due to the low performance of CPU, the speed maybe worse. 

And the example that you show is reasonable since it's pretty different from our training set.

**Comment by Honghe** [comment]
Among the total time, The line detect elapsed-time is 29.3s.

metric code:

```
# Get points and descriptors.
start1 = time.time()
lines = detector.detect(oriimg)
end1 = time.time()
print(f'Detect time elapsed {end1-start1:.1f}s')
```

output:

```
Detect time elapsed 29.3s
```

**Comment by Delay-Xili** [comment]
I think no bug there, that it is. The inference elapsed-time is normal.
